publish/maven-central.gradle

```diff
@@ -0,0 +1,47 @@
+apply plugin: 'signing'
+
+def property = { key ->
+  return (project.hasProperty(key) ? project."$key" : System.getenv(key) )
+}
+
+def user = property("MAVEN_CENTRAL_USER")
+def password = property("MAVEN_CENTRAL_PASSWORD")
+def signingKeyId = property("MAVEN_CENTRAL_SIGNING_KEY_ID")
+def signingPassword = property("MAVEN_CENTRAL_SIGNING_PASSWORD")
+def signingKeyRingFile = property("MAVEN_CENTRAL_SIGNING_KEY_RING_FILE")
+def isSnapshot = property("MAVEN_CENTRAL_IS_SNAPSHOT")
+
+publishing {
+  repositories {
+    maven {
+      it.name "MavenCentral${isSnapshot ? "Snapshot" : "Staging"}"
```

Comment:
[INQ] Why is this `Snapshot` or `Staging` only, should it be `Release`?

Reply:
Maven Central doesn't let you publish directly to the Release repository. You have to first publish to the Staging repo, and then promote it from Staging to Release from within the UI at https://s01.oss.sonatype.org/ . 

This is actually similar to how Bintray/JCenter worked: you had to publish to Bintray and then promote it to be "published" to JCenter. But on our Bintray publishing config, we were using a property appended to the API `publish=1` which automatically published it for us.

It is actually possible to promote from Staging to Release on CI using a Gradle plugin ([see here](https://getstream.io/blog/publishing-libraries-to-mavencentral-2021#automating-sonatype-actions)). But according to that article, the plugin can be buggy due to changing APIs, so I think it's better to promote it manually for now. Maybe in the future we can do it automatically if we find a better way.

Reply:
I see. Thank you for this information.